### DISCUSSION TITLE
Questions before buying Create3

### DISCUSSION THREAD

**Discussion by gmptrr98** [iRobotEducation/create3_docs#132] [Questions] [answered]
Hello, I'm working on my master thesis and I'll probably use Create3 as a case study.
Just a couple of questions:
1. What are the exact specifics of the IR obstacle sensors (distance, frequency etc...)? 
2. How does Create3 the find the docking station? Would it still find it if the dock was moved? What do the RCON sensor and the IR sensor window on the dock exactly do?
3. Has Create3 already some mapping capabilities without adding a Lidar or camera? How does it work?

If you could respond with as many details as possible that would be very helpful, thank you for the help!

**Comment by shamlian** [comment, marked as answer]
1. The IR obstacle sensors use 940 nm light and are AC coupled (external source lighting-insensitive) reflective sensors. Their range depends on surface quality of the target. Things that reflect IR are much more visible than things that do not (you can see a white wall further away than a black wall). I don't have exact ranges off hand, but to be clear, these sensors are not rangefinders, they are reflective obstacle sensors.

2. The Create 3 robot uses 38 kHz modulated infrared light emitted by the dock, and receivers on its bumper, to servo into the dock. The robot must be in view of the dock to get into the charger. The image on the [docs site](https://iroboteducation.github.io/create3_docs/api/docking/#docking-sensor-data) is probably helpful. You can either use a map you've generated to find the dock again, or else implement a "dock search" strategy by driving around and monitoring the /dock topic until it's in view, and then triggering a docking action.

3. The Create 3 robot is itself a platform and does not do the mapping for you, though it has all of the sensors required to do so, as it has all the same sensors as the Roomba i3. The robot will provide a wheel odometry + IMU + optical odometry fused `odom` coordinate system that can be used as input to a mapping system. Mapping system implementation is up to the learner, or else ROS 2 has many standard packages for mapping and navigation. One caveat of this latter option is that to my knowledge, all open-source ROS 2 mapping packages require additional sensors (either image or depth sensing, like a camera or a lidar).
